### PR TITLE
Safety Code

### PR DESCRIPTION
**Pull request by sahil-kale**

### CODE REVIEW

File: Common/Inc/Interchip.h [hunk 1]
```diff
@@ -2,12 +2,16 @@
...
 } Interchip_StoA_Packet;    //Safety to Autopilot packet
 
 typedef struct {
 	int16_t PWM[12];
 	uint16_t autonomous_level;
-} Interchip_AtoS_Packet;    //Autopilot to Safety packet
\ No newline at end of file
+} Interchip_AtoS_Packet;    //Autopilot to Safety packet
+
+#endif
```
Comment (Pop0097) [inline Common/Inc/Interchip.h:17]: Little last line thing

File: Safety/CMakeLists.txt [hunk 2]
```diff
@@ -11,22 +11,25 @@ add_definitions(
...
 # Tests
 #add_subdirectory(Tests)
-file(GLOB_RECURSE C_SOURCES  "../Common/Src/*.c" "../Common/Src/stm32/*.c" "Src/*.c" , "Libraries/Drivers/Src/stm32f0/*.c")
-file(GLOB_RECURSE CXX_SOURCES  "../Common/Src/*.cpp" "../Common/Src/stm32/*.cpp" "Src/*.cpp" , "Libraries/Drivers/Src/stm32f0/*.cpp")
+file(GLOB_RECURSE C_SOURCES  "boardfiles/Drivers/STM32F0xx_HAL_Driver/Src*.c" "boardfiles/Src/*.c")
+file(GLOB_RECURSE CXX_SOURCES "boardfiles/Src/*.cpp" "Src/*.cpp")
+
+
+if( EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/boardfiles/Src/main.c")
+  message( FATAL_ERROR "Hey Bub, you screwed up, but it's all good. You see, CubeMx generates a main.c rather than a main.cpp, so before generation, you need to change the extension of boardfiles/Src/main.cpp to boardfiles/Src/main.c, allowing cubeMX to update your main file. Then, after generation, rename the file to boardfiles/Src/main.cpp." )
```
Comment (Pop0097) [inline Safety/CMakeLists.txt:31]: I like this XD

File: Safety/Src/safety_controller.cpp [hunk 3]
```diff
@@ -0,0 +1,67 @@
+#include "safety_controller.hpp"
+#include "PWM.hpp"
+#include "stdlib.h"
+#include "stm32f0xx_hal_iwdg.h"
```
Comment (Pop0097) [inline Safety/Src/safety_controller.cpp]: I don't think we should be including any HAL files directly into .cpp files. I remember Anthony mentioning something about this to me.

File: Safety/Src/safety_controller.cpp [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+//#include "Interchip.h"
+#include "interchip_S.hpp"
```
Comment (Pop0097) [inline Safety/Src/safety_controller.cpp:5]: We can comment this out since interchip_S.hpp isn't even compiled

File: Autopilot/Inc/Interchip_A.h [hunk 5]
```diff
@@ -3,7 +3,8 @@
 
 #define INTERCHIP_TRANSMIT_DELAY 5
 
+void Interchip_Run(void const *argument);
 int16_t *Interchip_GetPWM(void);
 void Interchip_SetPWM(int16_t *data);
-uint16_t Interchip_GetSafetyLevel(void);
+uint16_t Interchip_GetAutonomousLevel(void);
 void Interchip_SetAutonomousLevel(uint16_t data);
```
Comment (Pop0097) [inline Autopilot/Inc/Interchip_A.h:10]: last line thingy

**Review by Pop0097 — COMMENTED** [review]
Didn't get a chance to look at `Safety/Libraries/Drivers/Src/stm32f0/stm32f0xx_hal_msp.c` and `Safety/Libraries/Drivers/Src/stm32f0/stm32f0xx_it.c `. Will look at them later. 

I didn't look at any files in `/boardfiles` since they appear to just be CubeMX generated. Please correct me if I am wrong.

**Review by Fberbari — APPROVED** [review]
Dhruv's comments are good. We'll address them. But it's critical this gets merged asap.

File: Safety/Src/safety_controller.cpp (hunk 3 above)
Comment (sahil-kale) [reply Safety/Src/safety_controller.cpp]: yeee you right, that was leftover code

File: Autopilot/Inc/Interchip_A.h (hunk 5 above)
Comment (sahil-kale) [reply Autopilot/Inc/Interchip_A.h:10]: That is autopilot bug not safety lmao

**Pull request merged**